python/src/aac/context/definition_parser.py

```diff
@@ -813,11 +813,16 @@ def load_definitions(
             for definition in self.parsed_definitions + self.context.get_definitions()
             if definition.get_root_key() == "primitive"
         }
-        self.fully_qualified_name_to_definition = {
-            definition.get_fully_qualified_name(): definition
-            for definition in self.parsed_definitions + self.context.get_definitions()
-            if "package" in definition.structure[definition.get_root_key()]
-        }
+        self.fully_qualified_name_to_definition = {}
+        for definition in self.parsed_definitions + self.context.get_definitions():
+            if "package" in definition.structure[definition.get_root_key()]:
+                fully_qualified_name = definition.get_fully_qualified_name()
+
+                # This is so requirements specifically do not get overwritten.  Although other definition types will still get overwritten, so we may need to find a better solution eventually.
```

Comment:
```suggestion
                # This is so requirements specifically do not get overwritten.  Although other definition types might still get overwritten, so we may need to find a better solution eventually.
```

Reply:
We don't know for certain that other definitions are being overwritten at this time, the combination of the package reference and name reference was part of mitigating that potential for definitions, but it wasn't accommodating requirement definitions being slightly different.

Reply:
They will get overwritten in the fully qualified name dictionary if they share a name.  I believe this only affects datarefs, so we just haven't run into it yet.  Its just that requirements are the most common dataref objects that could share a name.

Reply:
But the logic within the `load_definitions` method when it is assigning the key (our formatting of a name for referencing) for the dictionary is originally intending to use a combination of the `definition.package` and `definition.name` to help mitigate the name collision potential. 

Our requirement data structures do not have a concept of `package` to help ensure unique names, so we went a step further and for `req` definitions are appending the `req_id` portion as well now. 

And a `dataref` is intended to drill to a specific portion of the `definition` for referencing which generally is just a `name`, so we need to look into what version of `name` is being leveraged. Is it the `name` assigned to the `name` field in the definition or is it the `name` key created for the dictionary registry of loaded definitions.